sqlflow/env_expand.py

```diff
@@ -0,0 +1,51 @@
+import os
+import re
+from datetime import datetime, timedelta
+
+
+class EnvExpanderError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+    def __str__(self):
+        return repr(self.message)
+
+class EnvExpander(object):
+    def __init__(self, environ=os.environ):
+        self.environ = environ
+        self.pattern_env = re.compile("\$\{(.*?)\}")
+        # TODO(Yancey1989): support more date expression
+        self.pattern_date_expr = re.compile(r"(yyyymmdd)\Z|(yyyymmdd)\s(\+|\-)\s(\d+)d")
```

Comment:
should be `yyyyMMdd` with `MM` capitalized

Reply:
Seems there are several different standards:
- `yyyyMMdd` is Java idiomatic usage, ref [SimpleDataFoamt](https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html), this rule is used in Aliyun.
-  [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) uses  capitalized letter: `YYYY-MM-DD`
- Python datetime uses `%Y-%m-%d`

Reply:
@typhoonzero , I updated this PR and follows Java habit which used in Aliyun.